charts/cass-operator-chart/templates/deployment.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.deploymentName }}
```

Comment:
should these placeholders be named all specifically for cass operator, rather than leaving some of them generic? like `.Values.cassOperatorName` , `.Values.cassOperatorReplicas`, etc?

Reply:
The only thing I named with a cassOperator prefix was cassOperatorNamespace, because there is a built-in NAMESPACE value that was conflicting with it.  As far as prefixing the other values, I can do it, but I don't think it's necessary.

See:

https://devcards.io/smurf-naming-convention

Reply:
It looks like elastic uses a generic "Namespace" placeholder in their helm chart:

https://github.com/elastic/helm-charts/blob/master/elasticsearch/templates/rolebinding.yaml#L19

Having said that, most of their resources do not seem to have a namespace set at all.

Reply:
yeah I think `namespace` might be a global thing for helm, kinda like it is for `kubectl`

Reply:
Regarding using a namespace, does it seem like this best practice is still valid? It _seems_ different from the approach we are using, but I may be misunderstanding:
https://github.com/helm/helm/issues/5465#issuecomment-473942223

Reply:
There's a lot of back and forth in that thread, including a bunch of "confused" emojis on the comment you referenced.

Some people believe that this special-case treatment of namespace breaks configuration as code:

https://github.com/helm/helm/issues/5465#issuecomment-548247929

I think it breaks the "principal of least surprise" to treat namespace different than every other parameter, when it could just be handled by the default behavior.

Reply:
https://github.com/helm/helm/issues/5465#issuecomment-573152021 I think the followup comment by the maintainer hits the point. They tried to make this change, and the community rejected it. There is a strong training effect with kubectl, where people are frequently specifying namespaces as arguments, so I think from that perspective "principal of least surprise" works on the side of specifying it as a helm argument. That said, I'm fine with your changes, and certainly not trying to block traffic here.

Reply:
Yeah, if this is what is standard practice, then I am fine with changing it.  Does anyone feel strongly either way?

Reply:
Ok I researched this and it's very common to to use the generic helm namespace support...

https://cilium.readthedocs.io/en/stable/gettingstarted/k8s-install-gke/
https://istio.io/docs/setup/install/helm/#installation-steps
https://pingcap.com/docs/tidb-in-kubernetes/stable/deploy-tidb-from-kubernetes-gke/#deploy-tidb-operator

Linkerd adding a large explanation about why they couldn't support it, but it's confusing

https://linkerd.io/2/tasks/install-helm/#customizing-the-namespace

Mysql Operator made their own config 🤔 

https://github.com/oracle/mysql-operator/blob/master/docs/tutorial.md#configuration